scripts/ci/check_compliance.py

```diff
@@ -676,6 +677,80 @@ def get_logging_syms(self, kconf):
 
         return set(kconf_syms)
 
+    def check_banned_defconfigs(self, kconf):
+        """
+        Checks that there are no banned Kconfigs used in board defconfig files
+        """
+        # Grep for symbol references.
+        #
+        # Example output line for a reference to CONFIG_FOO at line 17 of
+        # foo/bar.c:
+        #
+        #   foo/bar.c<null>17<null>#ifdef CONFIG_FOO
+        #
+        # 'git grep --only-matching' would get rid of the surrounding context
+        # ('#ifdef '), but it was added fairly recently (second half of 2018),
+        # so we extract the references from each line ourselves instead.
+        #
+        # The regex uses word boundaries (\b) to isolate the reference, and
+        # negative lookahead to automatically whitelist the following:
+        #
+        #  - ##, for token pasting (CONFIG_FOO_##X)
+        #
+        #  - $, e.g. for CMake variable expansion (CONFIG_FOO_${VAR})
+        #
+        #  - @, e.g. for CMakes's configure_file() (CONFIG_FOO_@VAR@)
+        #
+        #  - {, e.g. for Python scripts ("CONFIG_FOO_{}_BAR".format(...)")
+        #
+        #  - *, meant for comments like '#endif /* CONFIG_FOO_* */
+
+        banned_symbols = [
+            "PINCTRL",
+        ]
+
+        # Maps each undefined symbol to a list <filename>:<linenr> strings
+        banned_to_locs = collections.defaultdict(list)
+
+        # Warning: Needs to work with both --perl-regexp and the 're' module
+        regex = r"\bCONFIG_[A-Z0-9_]+\b(?!\s*##|[$@{(.*])"
+
+        # Skip doc/releases and doc/security/vulnerabilities.rst, which often
+        # reference removed symbols
```

Comment:
the actual implementation is more like: "only search in `boards/` though, right? I'd mention that in the comment

Reply:
The base for this was duplicated from the Kconfig check, overlooked removing that comment